### PR TITLE
Captain can buy a new escape shuttle

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Using the communications console. This uses cargo points.

Some (small, awful) shuttles cost negative points, granting a bonus to cargo if you purchase them.

Like everything else, the costs are all pretty much made up with little thought.

I need to clean up the code a bit, but it all functions.

### CODE REVIEW

**Comment by PKPenguin321** [comment]
if you "buy" a small awful shuttle then invest earned money in stocks and make new money can you undo the buy?

File: code/datums/shuttles.dm [hunk 1]
```diff
@@ -185,6 +198,7 @@
 	admin_notes = "Currently larger than ferry docking port on Box, will not \
 		hit anything, but must be force docked. Trader and ERT bodyguards are \
 		not included."
+	credit_cost = 7500
```
Comment (ChangelingRain) [inline code/datums/shuttles.dm]: This isn't an emergency shuttle!

File: code/datums/shuttles.dm [hunk 2]
```diff
@@ -171,6 +183,7 @@
 		definitely no zombifyin' reagents!"
 	admin_notes = "Meat currently contains no zombifying reagents, lizard on \
 		meatspike must be spawned in."
+	credit_cost = 5000
```
Comment (ChangelingRain) [inline code/datums/shuttles.dm]: Neither is this!

File: code/datums/shuttles.dm (hunk 1 above)
Comment (KorPhaeron) [inline code/datums/shuttles.dm]: oops

**Comment by KorPhaeron** [comment]
@PKPenguin321 

Can only buy one shuttle per round

**Comment by kevinz000** [comment]
Time to buy supermatter shuttle.

**Comment by KorPhaeron** [comment]
Not an option, sorry. It costs infinity credits

**Comment by kevinz000** [comment]
;_;

**Comment by ma44** [comment]
@KorPhaeron it would be possible if you exported lemons

**Comment by PKPenguin321** [comment]
kys

**Comment by Shadowlight213** [comment]
>ma44 said the L word.

B&

**Comment by Core0verload** [comment]
Can I by NO SHUTTLE for -10000 credits? I want crew to build one before fleeing.

**Comment by KorPhaeron** [comment]
@Core0verload 

You can buy an empty one without walls for -7500, so sorta yeah

**Comment by RemieRichards** [comment]
@KorPhaeron actually you probably could buy a shuttle for Infinity, given how broken stock markets are atm, and the fact our Infinity is just a big number.

File: code/game/machinery/computer/communications.dm [hunk 3]
```diff
@@ -130,6 +131,31 @@ var/const/CALL_SHUTTLE_REASON_LENGTH = 12
...
+		if("purchase_menu")
+			state = STATE_PURCHASE
+
+		if("buyshuttle")
+			if(authenticated==2)
+				var/datum/map_template/shuttle/S = locate(href_list["chosen_shuttle"])
+				if(S)
+					for(var/obj/machinery/shuttle_manipulator/M in machines)
+						if(M.shuttle_purchased)
+
```
Comment (RemieRichards) [inline code/game/machinery/computer/communications.dm]: ?

File: code/game/machinery/computer/communications.dm [hunk 4]
```diff
@@ -463,6 +490,15 @@ var/const/CALL_SHUTTLE_REASON_LENGTH = 12
 			else
 				dat += "<b>Emergency Maintenance Access is currently <font color='green'>DISABLED</font></b>"
 				dat += "<BR>Lift access restrictions on maintenance and external airlocks? <BR>\[ <A HREF='?src=\ref[src];operation=enableemergency'>OK</A> | <A HREF='?src=\ref[src];operation=viewmessage'>Cancel</A> \]"
+		if(STATE_PURCHASE)
+			dat += "Budget: [SSshuttle.points] Credits.<BR>"
+			for(var/shuttle_id in shuttle_templates)
+				var/datum/map_template/shuttle/S = shuttle_templates[shuttle_id]
+				if(S.credit_cost < INFINITY)
```
Comment (RemieRichards) [inline code/game/machinery/computer/communications.dm]: ... or you cheat and make infinity explicitly impossible, that works too.

Comment (KorPhaeron) [inline code/game/machinery/computer/communications.dm]: Seemed like the best way to stop the list filling up with unbuyable shuttles while saving another var

File: code/game/machinery/computer/communications.dm [hunk 5]
```diff
@@ -629,3 +665,14 @@ var/const/CALL_SHUTTLE_REASON_LENGTH = 12
 /obj/machinery/computer/communications/proc/overrideCooldown()
 	var/obj/item/weapon/circuitboard/computer/communications/CM = circuit
 	CM.lastTimeUsed = 0
+
+
+/obj/machinery/computer/communications/proc/get_shuttles_for_sale()
+	var/dat = ""
```
Comment (RemieRichards) [inline code/game/machinery/computer/communications.dm]: make this a list...

File: code/game/machinery/computer/communications.dm [hunk 6]
```diff
@@ -629,3 +665,14 @@ var/const/CALL_SHUTTLE_REASON_LENGTH = 12
...
+
+/obj/machinery/computer/communications/proc/get_shuttles_for_sale()
+	var/dat = ""
+	for(var/datum/map_template/shuttle/emergency/E in shuttle_templates)
+		var/datum/map_template/shuttle/S = E
+
+		dat += "[S.name] | [S.credit_cost] Credits<BR>"
+		dat += "[S.description]<BR>"
+		dat += "<A href='?src=\ref[src];buyshuttle=\ref[S]'>(<font color=red><i>Purchase</i></font>)</A>"
+	return dat
```
Comment (RemieRichards) [inline code/game/machinery/computer/communications.dm]: and the return ```the_list.Join("")```
it's faster as you don't bother the string table as much.

**Comment by optimumtact** [comment]
Another "adminonly" feature creeping into the game like clockwork

File: code/game/machinery/computer/communications.dm (hunk 4 above)
Comment (RemieRichards) [inline code/game/machinery/computer/communications.dm]: yeah no it's fine, it's just I made a point about broken stock markets allowing the purchase of the INFINITY ships, but this works around that.

**Comment by KorPhaeron** [comment]
@optimumtact 

Are you saying I should add it to science instead?

**Comment by Saegrimr** [comment]
We gonna start bwoinking people for buying shuttles that people can't survive in by default?

**Comment by KorPhaeron** [comment]
@Saegrimr 

I  thought better of it and disabled all the lethal ones (though buying a small one will probably still cause death)

**Comment by Incoming5643** [comment]
The landing zones on some stations are too small to take the larger shuttles, if this happens the round cannot end without admin intervention

**Comment by optimumtact** [comment]
Any timeline on finishing this?

**Comment by KorPhaeron** [comment]
couple days probably, I got distracted by gun code tonight

**Comment by anconfuzedrock** [comment]
wheres the goof shuttle

**Comment by KorPhaeron** [comment]
busy

**Comment by Iamgoofball** [comment]
:(

**Pull request closed without merging**